homeassistant/components/wake_word/__init__.py

```diff
@@ -0,0 +1,170 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+from .models import DetectionResult, WakeWord
+
+__all__ = [
+    "async_default_engine",
+    "async_get_wake_word_detection_entity",
+    "DetectionResult",
+    "DOMAIN",
+    "WakeWord",
+    "WakeWordDetectionEntity",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)
+
+
+@callback
+def async_default_engine(hass: HomeAssistant) -> str | None:
+    """Return the domain or entity id of the default engine."""
+    return next(iter(hass.states.async_entity_ids(DOMAIN)), None)
+
+
+@callback
+def async_get_wake_word_detection_entity(
+    hass: HomeAssistant, entity_id: str
+) -> WakeWordDetectionEntity | None:
+    """Return wwd entity."""
+    component: EntityComponent = hass.data[DOMAIN]
+
+    return component.get_entity(entity_id)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up STT."""
+    component = hass.data[DOMAIN] = EntityComponent(_LOGGER, DOMAIN, hass)
+    component.register_shutdown()
+
+    # Register websocket API
+    websocket_api.async_register_command(hass, websocket_detect)
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    component: EntityComponent = hass.data[DOMAIN]
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    component: EntityComponent = hass.data[DOMAIN]
+    return await component.async_unload_entry(entry)
+
+
+class WakeWordDetectionEntity(RestoreEntity):
+    """Represent a single WWD provider."""
+
+    _attr_should_poll = False
+
+    @property
+    @abstractmethod
+    def supported_wake_words(self) -> list[WakeWord]:
+        """Return a list of supported wake words."""
+
+    @abstractmethod
+    async def async_process_audio_stream(
+        self, stream: AsyncIterable[tuple[bytes, int]]
+    ) -> DetectionResult | None:
+        """Try to detect wake word(s) in an audio stream with timestamps.
+
+        Audio must be 16Khz sample rate with 16-bit mono PCM samples.
+        """
+
+
+@websocket_api.websocket_command(
+    vol.All(
+        websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend(
+            {
+                vol.Required("type"): "wake_word/detect",
+                vol.Optional("entity_id"): vol.Any(str, None),
+                vol.Optional("timestamp_start"): vol.Any(int, None),
+            },
+        ),
+    ),
+)
+@websocket_api.async_response
+async def websocket_detect(
```

Comment:
This will not be necessary if we expose this via the Assist pipeline run WS command.